haskellPackages: stackage LTS 23.21 -> LTS 23.24

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `staging`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `staging` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #404362. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

alexfmpe committed:
haskellPackages.ghc: don't depend on libiconv on android-prebuilt

Fixes build of `pkgsCross.aarch64-android-prebuilt.haskell.packages.ghc910.primitive`:
```
aarch64-unknown-linux-android-ld: error: attempted static link of dynamic object /nix/store/dc4sk70a32x7qzvqr81rq9036cslrlw3-libiconv-aarch64-unknown-linux-android-1.17/lib/libiconv.so
<no location info>: error:
    `aarch64-unknown-linux-android-ld' failed in phase `Merge objects'. (Exit code: 1)
```

alexfmpe committed:
haskellPackages: disable shared libraries on android-prebuilt

Fixes build of `pkgsCross.aarch64-android-prebuilt.haskell.packages.ghc910.attoparsec`:
```
RPATH of binary /nix/store/g6ncgk2b04mv722hh2jgfbixyawbwabv-attoparsec-aarch64-unknown-linux-android-0.14.4/lib/ghc-9.10.1/lib/aarch64-android-ghc-9.10.1-inplace/libHSattoparsec-0.14.4-GkMCGqk5eKArKmd1r9IB7-ghc9.10.1.so contains a forbidden reference to /build/
```

sternenseemann committed:
haskell.packages.ghc912.feed: jailbreak t avoid bounds on time, base

sternenseemann committed:
haskell.packages.ghc912.xml-conduit: fix test suite with GHC 9.12

neilmayhew committed:
haskellPackages.cabal-lenses: build with required version of Cabal

Also remove broken status

ozkutuk committed:
haskellPackages.migrant-{core,sqlite-simple,hdbc,postgresql-simple}: unbreak

sternenseemann committed:
haskell.compiler.ghc910: 9.10.1 -> 9.10.2

sternenseemann committed:
haskell.packages.ghc910.git-annex: test on Hydra

I'd like to keep this job until we upgrade to the next Stackage LTS
version, so we know that it's possible to build git-annex without
filepath-bytestring.

alexfmpe committed:
haskellPackages.android-activity: unmark broken on android

alexfmpe committed:
haskellPackages: add aarch64-android-prebuilt to release-haskell jobset on ghc 9.10

nixpkgs-ci[bot] committed:
Merge 11e0819a5c2ff8b6e1a060f9b34c2516358930f0 into haskell-updates

sternenseemann committed:
haskell.packages.ghc90.sv2v: drop obsolete override

See #407367.

nixpkgs-ci[bot] committed:
Merge e3beaa67c475589eb5d7e8d21f45308eedf63dec into haskell-updates

wolfgangwalther committed:
haskellPackages.migrant-{core,sqlite-simple,hdbc,postgresql-simple}: unbreak (#407430)

nixpkgs-ci[bot] committed:
Merge dc2d5f03912aebe6f495e61b5347f104a21bad1b into haskell-updates

nixpkgs-ci[bot] committed:
Merge 5c68d925b545ab6daea44ba1295756bb2493c6af into haskell-updates

akshaymankar committed:
haskellPackages.amazonka-s3-streaming: Use source from hackage

This package is not part of the amazonka repository.

Co-authored-by: sternenseemann <sternenseemann@systemli.org>

nixpkgs-ci[bot] committed:
Merge 873183508cf8cdb6751f24d31fc01122fd1797f1 into haskell-updates

nixpkgs-ci[bot] committed:
Merge 89eac1578109c167c46ec3d4ca0f1d133753be81 into haskell-updates

maralorn committed:
haskellPackages.cabal-lenses: build with required version of Cabal (#406549)

nixpkgs-ci[bot] committed:
Merge fe51d34885f7b5e3e7b59572796e1bcb427eccb1 into haskell-updates

sternenseemann committed:
haskell.compiler.ghc94: don't roundtrip C compilation via assembly

GHC can be used to compile C sources which causes it to drive the
configured C compiler. This is particularly relevant during GHC's own
compilation, e.g. when building the rts. GHC takes a peculiar approach,
always generating intermediate assembly instead of letting the C
compiler emit object files directly. This causes an assembler check in
clang >= 18 to fail on rts/StgCRun.c, failing the GHC build on darwin
completely.

Later GHC versions don't exhibit this issue since object code is emitted
directly since 9.6. The easiest way to resolve the compilation failure
seems to backport this change. However, the patch only applies on
GHC 9.4 which is done here. The patch may be difficult to backport
further, as the changed code was extensively refactored in GHC 9.4.

Reference #367686.

nixpkgs-ci[bot] committed:
Merge 980ae9854c3ef2b0a65d5faee6a7a5b2f9e15ac2 into haskell-updates

nixpkgs-ci[bot] committed:
Merge a61befb69a171c7fe6fb141fca18e40624d7f55f into haskell-updates

nixpkgs-ci[bot] committed:
Merge 8158addeaeb69921e8b474a81fafb7d97ded2850 into haskell-updates

nixpkgs-ci[bot] committed:
Merge 3108eaa516ae22c2360928589731a4f1581526ef into haskell-updates

nixpkgs-ci[bot] committed:
Merge d9dd002a294e676c49e27eae9166f73bd2ae5684 into haskell-updates

sternenseemann committed:
haskell.compiler.ghc*Binary: detect -lnuma based on library list

This is more robust when new bindists are added to expressions since no
additional conditional needs to be updated.

sternenseemann committed:
haskell.compiler.ghc902Binary: init at 9.0.2

We can/should use 9.0.2 to bootstrap some GHC versions, having the
bindist makes things easier in some cases.

- To bootstrap GHC 9.4 using 9.2 is not possible and 8.10.7 doesn't
  always work. Since source built 9.0.2 is broken on darwin at the
  moment, the bindist helps. Reference #367686, #410085.
- To bootstrap GHC 9.2 without GHC 8.10. Reference #346214.

sternenseemann committed:
haskell.compiler.ghc94: bootstrap from 9.0.2 bindist

nixpkgs-ci[bot] committed:
Merge 3f62a1ca1a6ff1b7c21e61ad66f72f0cd4925301 into haskell-updates

sternenseemann committed:
haskell.compiler.ghc928: bootstrap from 9.0.2 bindist

nixpkgs-ci[bot] committed:
Merge ee2477b0b50e045c6e4e4ea66b69bb803aed6040 into haskell-updates

nixpkgs-ci[bot] committed:
Merge 4420d42422ac6a7994e3f897a3e54693a6db0194 into haskell-updates

sternenseemann committed:
Merge branch staging-next into haskell-updates

nixpkgs-ci[bot] committed:
Merge c936d57413b770679543a3aa85791f299b931a7b into haskell-updates

sternenseemann committed:
Merge branch staging-next into haskell-updates

sternenseemann committed:
haskellPackages: stackage LTS 23.21 -> LTS 23.24

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

sternenseemann committed:
all-cabal-hashes: 2025-05-05T12:06:43Z -> 2025-06-01T18:10:16Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

nixpkgs-ci[bot] committed:
Merge f9969bf7f6a6ab7e0fadd889ddd9fa6ad8cc5288 into haskell-updates

wolfgangwalther committed:
haskell.packages.ghc910.ghc-lib: 9.10.2.20250503 -> 9.10.2.20250515

Fixes eval.

maralorn committed:
pandoc: Backport patch from 3.6.4 to fix tests

maralorn committed:
haskell-language-server: Fix build (on ghc 9.12)

sternenseemann committed:
elmPackages.elm-instrument: build using GHC 9.4

As Emily has pointed out, elm-instrument builds with GHC 9.4 without any
modifications. Later versions of GHC won't work due to the pinned
indents package version which is affected by a breaking change in some
core package.

sternenseemann committed:
haskellPackages.bsb-http-chunked: re-enable tests on i686

The corresponding bug in blaze-builder has been fixed.

wolfgangwalther committed:
haskellPackages.feedback: drop override

This has been updated to 0.2.0.0, but it's still marked as broken (and
rightly so).

Clearing the assert.

wolfgangwalther committed:
haskellPackages.gogol: remove source override

1.0.0.0 has now been released on hackage.

wolfgangwalther committed:
haskellPackages.gi-vte: drop patch

We have 2.91.35+ now, so the patch is included.

wolfgangwalther committed:
haskellPackages: use warnIf instead of assert

Contrary to intuition, an assert will **not** fail CI's Eval workflow.
Instead, the package will be hidden from Eval. A warning will be
evaluated - and then cause Eval to fail.

Also, all warnings are printed, before Eval fails. Thus we can see at
one glance which packages need to be checked and don't need to iterate.

wolfgangwalther committed:
haskellPackages: use warnIf instead of assert (#413299)

alexfmpe committed:
haskellPackages.jsaddle-warp: fix build and tests

alexfmpe committed:
haskellPackages: remove patches merged upstream

wolfgangwalther committed:
haskellPackages: remove patches merged upstream (#413367)

wolfgangwalther committed:
haskellPackages: unbreak packages

This commit has been generated by

maintainers/scripts/haskell/unbreak.nu

and

maintainers/scripts/haskell/regenerate-hackage-packages.sh

wolfgangwalther committed:
haskellPackages: run treefmt

Introduced in previous commit.

alexfmpe committed:
haskellPackages: remove unneeded jailbreaks

sternenseemann committed:
haskellPackages.persistent-test: pin to < 2.13.1.4, match persistent

Since Stackage LTS doesn't run test suites, PVP legal updates can
sometimes break test suites. This happened here. See the comment for
details.

Presumably because of configuration ordering, there's no elegant way to
override a constraint from stackage.yaml at the moment.

sternenseemann committed:
git-annex: update sha256 for 10.20250520

Update the link to the mentioned issue and summarize upstream's
response.

cdepillabout committed:
[haskell-updates] haskellPackages.persistent-test: pin to < 2.13.1.4, match persistent  (#413885)

maralorn committed:
haskellPackages: fixes for pkgsCross.aarch64-android-prebuilt (#355543)

sternenseemann committed:
haskell.packages.ghc9102.servant-client: fix for text >= 2.1.2

The issue presents itself in README.lhs which normally is a symlink to
README.md. Upstream has a patch fixing the issue, but we need to
additionally deal with the fact that the Hackage tarball doesn't
preserve the symlink, of course.

wolfgangwalther committed:
haskellPackages.jsaddle-warp: fix build and tests (#413334)

sternenseemann committed:
Merge 73557c22ecaaa426f65dfd7bd18467dc451ebedd into haskell-updates

nixpkgs-ci[bot] committed:
Merge 047f31d25b0dc7b55587e75d6f5cee4922262d24 into haskell-updates

rnhmjoj committed:
haskellPackages.adblock2privoxy: unbreak

wolfgangwalther committed:
haskellPackages.adblock2privoxy: unbreak (#415307)

maralorn committed:
pandoc-crossref: Remove myself as maintainer

maralorn committed:
haskell.packages.ghc910.haskell-language-server: Fix build

sternenseemann committed:
haskellPackages.pandoc-crossref: pin to 0.3.19

This matches pandoc == 3.6.*.

turboMaCk committed:
elmPackages.elm-instrument: build using GHC 9.4 (#413234)

wolfgangwalther committed:
elmPackages.elm-coverage: drop

This allows dropping elmi-to-json, which in turn allows dropping GHC
8.10.7. Along with elmi-to-json, we can remove the patchBinwrap utility,
which was only used for that purpose.

Along with elm-coverage, we can also drop elm-instrument, which was only
used as a dependency of the former, and it's reverse dependency indents.

wolfgangwalther committed:
elmPackages: run generate-node-packages.sh

I did not commit changes to node-composition.nix, which didn't seem
intended.

wolfgangwalther committed:
elmPackages.elm-pages: fix build

wolfgangwalther committed:
elmPackages.elm-verify-examples: fix build

elm-verify-examples doesn't depend on elmi-to-json anymore.

wolfgangwalther committed:
elmPackages.elm-verify-examples: remove unused dependency esbuild

wolfgangwalther committed:
elmPackages.elm-coverage: drop (#415911)

alexfmpe commented:
> We should look over the remaining failures and look into merging soon.

We only have the ghc-specific HLS failures to consider because all others maintained failures depend on `jsaddle-webkit2gtk -> gi-webkit2 -> gi-soup2`.  FWIW this dependency isn't present when building miso/reflex-dom/etc with `pkgsCross.ghcjs.haskell.packages.ghc912`

sternenseemann commented:
w.r.t. gi-soup2

73b1018231d94b9a34871c622b6af2fde6dfafc0 is the first bad commit
```
commit 73b1018231d94b9a34871c622b6af2fde6dfafc0
Author: Jan Tojnar <jtojnar@gmail.com>
Date:   Mon Mar 3 00:02:15 2025 +0000

    gobject-introspection: 1.82.0 → 1.83.2
    
    https://gitlab.gnome.org/GNOME/gobject-introspection/-/compare/1.82.0...1.83.2
    
    Changelog-Reviewed-By: Jan Tojnar <jtojnar@gmail.com>

 ...-scanner-Prefer-some-getters-over-others.patch | 92 +++++++++++++++++++++
 .../libraries/gobject-introspection/default.nix   | 14 ++--
 2 files changed, 97 insertions(+), 9 deletions(-)
 create mode 100644 pkgs/development/libraries/gobject-introspection/0001-scanner-Prefer-some-getters-over-others.patch
```

sternenseemann committed:
Merge commit b8d24418ab34eaceb4dd9f3f2d0072fa3 into haskell-updates

sternenseemann committed:
haskell.packages.ghc94.stan: don't use directory-ospath-streaming

The reliance on the OsPath API of this package means that there's no
(easy?) install plan for GHC < 9.6. We only _need_ stan for GHC 9.4 HLS,
on lower versions it seems to be broken by independent issues.

wolfgangwalther committed:
haskell.packages.ghc94.stan: don't use directory-ospath-streaming (#418716)

JulienMalka committed:
haskellPackages.hexstring: unbreak

wolfgangwalther committed:
haskellPackages.postgrest: 13.0.0 -> 13.0.4

Just fixes some regressions in v13.

Changelog:
https://github.com/PostgREST/postgrest/blob/v13.0.4/CHANGELOG.md

wolfgangwalther committed:
haskellPackages.postgrest: 13.0.0 -> 13.0.4 (#418832)

nixpkgs-ci[bot] committed:
Merge e090b4f42a1590342b3e9babcef9c6717fd894ac into haskell-updates

wolfgangwalther reviewed (approved):
I think we fixed the HLS issues that were still open, so we should be good to go for now.

nixpkgs-ci[bot] committed:
Merge 4585b48c6f02d5a712559391ffa87017bd46dbee into haskell-updates

maralorn committed:
haskellPackages.hexstring: unbreak (#418719)

nixpkgs-ci[bot] committed:
Merge 767223d29de8e2baa5124a6f039fa264a9f4e0c5 into haskell-updates

sandydoo committed:
elmPackages.elm-format: 0.8.7 -> 0.8.8

This release officially supports GHC 9.6.

To avoid dependency conflicts with the elm compiler (see ansi-wl-pprint), elm-format is now built with GHC 9.8.

The GHC 9.4 elm package overrides have been removed.

sandydoo committed:
elmPackages: update and run the `update.sh` script

- Vendored the post-patch script for elm-format.
- Updated `update.sh` and committed the changes.

wolfgangwalther committed:
elmPackages.elm-format: 0.8.7 -> 0.8.8 (#419302)

sternenseemann committed:
haskellPackages.jpeg-turbo: fix dep discovery and tests

nixpkgs-ci[bot] committed:
Merge d079709547d8ee90ce524ce59cd2b7953de22987 into haskell-updates

nixpkgs-ci[bot] committed:
Merge 657b33de3db408c2023043a0ee21d61802cc90cf into haskell-updates

nixpkgs-ci[bot] committed:
Merge eddb0f04e854472ad6b9b24002b2dd9d336b5ee2 into haskell-updates

sternenseemann commented:
@rnhmjoj https://hydra.nixos.org/build/301060500

nixpkgs-ci[bot] committed:
Merge 9be32fbfaab592ccc39eb2f299e42aafddc26b37 into haskell-updates

nixpkgs-ci[bot] committed:
Merge c0874d923d0fd19d73d95a02820acd27ae11e320 into haskell-updates

sternenseemann committed:
haskellPackages.env-extra: fix flaky test failures

nixpkgs-ci[bot] committed:
Merge 668b089f3418d09158ea72075b9819ac3a6560dc into haskell-updates

lelambda committed:
haskellPackages.haskore: fixed broken package (doJailbreak, dontCheck)

peterbecich committed:
haskellPackages.cabal2nix: unstable-2025-04-30 -> unstable-2025-06-14

This update fixes the `regenerate-hackage-packages` script.

```
./maintainers/scripts/haskell/update-cabal2nix-unstable.sh
```

Previously, `regenerate-hackage-packages` failed with:
```
...
hackage2nix: user error (cannot parse cabal file /nix/store/kmz1d73l5k1nhv5269dl5i68yjwlbcir-unpacked-cabal-hashes/pms-application-service/0.0.3.0/pms-application-service.cabal)
hackage2nix: thread blocked indefinitely in an MVar operation
```

rnhmjoj committed:
haskellPackages.selda-json: unbreak

maralorn committed:
haskellPackages.haskore: fixed broken package (doJailbreak, dontCheck) (#419877)

peterbecich committed:
haskellPackages: regenerate package set based on current config

```
./maintainers/scripts/haskell/regenerate-hackage-packages.sh
```

No update of Hackage or Stackage.

nixpkgs-ci[bot] committed:
Merge 0b412937e24c7e166e3164b12e97144eef984697 into haskell-updates

maralorn committed:
fix `regenerate-hackage-packages` (#421127)

sternenseemann committed:
haskellPackages.cabal-cargs: remove dependence on location of source

sternenseemann committed:
haskellPackages.geomancy-layout: disable tests due to missing data

sternenseemann committed:
haskellPackages.cabal-cargs: enable separate bin output

wolfgangwalther committed:
haskellPackages.env-extra: fix flaky test failures (#420873)

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1816057](https://hydra.nixos.org/eval/1816057) of nixpkgs commit [028a94b](https://github.com/NixOS/nixpkgs/commits/028a94be88e833c919631df2973a00ec008062a8) as of 2025-06-23 09:22 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

sternenseemann committed:
maintainers/scripts/haskell: don't redundantly add C++ Nix to env

Since the shebang calls nix-shell, we can safely assume that Nix (Lix,
C++ Nix) is installed. Our scripts should support a wide enough range of
Nix versions so that using the “impure” version of the tool is not a
problem.

This works around #400784. My theory is that the Nix frontend commands
no longer work with older versions of the Nix daemon nor the Lix daemon
in our workloads.

wolfgangwalther committed:
maintainers/scripts/haskell: don't redundantly add C++ Nix to env (#421261)

nixpkgs-ci[bot] committed:
Merge e1db604e23f47297c241f315b8dd0955e0698cc3 into haskell-updates

sternenseemann committed:
haskellPackages.JuicyPixels-jpeg-turbo: also disable tests

This package is affected by the same problem as jpeg-turbo,
missing test files in the sdist.

sternenseemann committed:
top-level/release-haskell.nix: nixfmt attribute is just an alias now

sternenseemann committed:
haskell.compiler.ghc902Binary: workaround com.apple.provenance xattr

See #413450.

(Ported from 759f22430c78267e863e583c79dd2269b51d1e5c.)

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1816563](https://hydra.nixos.org/eval/1816563) of nixpkgs commit [bbb13da](https://github.com/NixOS/nixpkgs/commits/bbb13daacc156a67eb57d26fa2a78ef94fbac7f2) as of 2025-07-01 22:36 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates
